src/gaps/RetrievesFinder.ts

```diff
@@ -135,57 +175,53 @@ export function findRetrieves(
           expressionStack: [...expressionStack],
           path: exprRet.codeProperty
         });
-        withErrors.push(...withErrors);
       }
     }
   } else if (expr.type === 'Query') {
-    // Queries have the source array containing the expressions
-    (expr as ELMQuery).source?.forEach(s => {
-      const retrieves = findRetrieves(
-        elm,
-        allELM,
-        s.expression,
-        (expr as ELMQuery).localId,
-        valueComparisonLocalId,
-        [...expressionStack],
-        [...withErrors]
-      );
-      results.push(...retrieves.results);
-      withErrors.push(...retrieves.withErrors);
+    // For queries, recurse on all cases that may contain any ELM expression
+    // NOTE: ignoring "aggregate" for now, as we have no precedent for its use within queries for eCQMs
+    const query = expr as ELMQuery;
+
+    // Overwrite queryLocalId with new ID of the current query expression
+    const recursiveOpts: RecursiveCallOptions = {
+      ...defaultRecursiveOpts,
+      newQueryLocalId: query.localId
+    };
+
+    query.source?.forEach(s => {
```

Comment:
```suggestion
    query.source.forEach(s => {
```
I think we can ditch this question mark since source is 1...*